core/templates/pages/admin-page/misc-tab/admin-misc-tab.directive.html

```diff
@@ -72,6 +72,25 @@ <h3>Send a Test Mail to Admin</h3>
   </div>
 </md-card>
 
+<md-card class="oppia-admin-page-card oppia-page-card oppia-long-text">
+  <h3>Update username</h3>
+  <div class="oppia-query-form">
+    <form class="form" name="updateUsernameForm" ng-submit="$ctrl.updateUsername()">
+      <div class="form-group row">
+        <label>Old username:</label>
+        <input type="text" ng-model="$ctrl.oldUsername" required="true">
+      </div>
+      <div class="form-group row">
+        <label>New username:</label>
+        <input type="text" ng-model="$ctrl.newUsername" required="true" ng-maxlength="$ctrl.MAX_USERNAME_LENGTH">
```

Comment:
`ng-maxlength` will prevent users to type not more than 30 chars? (AFAIK it won't)

Either user maxlength or ng-maxlength with a proper warning message.

Reply:
You're right, `ng-maxlength` doesn't prevent the user from typing more than the threshold which is why I added the conditional disable to the submit button. 
But, I've changed this to `maxlength` which does prevent the user from entering more than 30 chars.